connector/connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala

```diff
@@ -46,6 +51,37 @@ class SparkConnectProtoSuite extends PlanTest with SparkConnectPlanTest {
     comparePlans(connectPlan.analyze, sparkPlan.analyze, false)
   }
 
+  test("Basic joins with different join types") {
+    val connectPlan = {
```

Comment:
can you add a test-case for unspecified as well to see that we catch the error?

Reply:
I discussed with @cloud-fan and we decided to remove bot `JoinType.CROSS` and `JoinType.Unspecified`. 

Proto is our API. We should make the proto itself less ambiguous. For example any proto submitted to the server should not contain a `JoinType.Unspecified`. It is either set a join type with explicit semantic (inner, left outer, etc.) or not set, which Spark by default treats as inner join.

Reply:
The reason for unspecified is not the proto contract but the language behavior for different auto generated targets. To avoid issues with defaults, the recommendation in the typical proto style guides is to always have the first element of an enum be unspecified. 

Cc @cloud-fan

Reply:
If the proto is an API, I'd say the join type is a required field and clients must set the join type in the join plan. For the python client, its dataframe API can omit the join type, and the python client should use INNER as the default join type.

Reply:
Ah if this is the style guide of protobuf, let's keep it.

Reply:
And the server can simply fail if it sees `UNSPECIFIED` join type?

Reply:
Yes exactly.